Wrappers/d3d8/IDirect3DDevice8.cpp

```diff
@@ -382,17 +382,13 @@ HRESULT m_IDirect3DDevice8::SetRenderState(D3DRENDERSTATETYPE State, DWORD Value
 	{
 		Value = (Value * 15) / 16;
 	}
-	
+
 	// Restores self shadows
-	if (EnableSelfShadows && State == D3DRS_STENCILPASS && (Value == D3DSTENCILOP_ZERO || Value == D3DSTENCILOP_REPLACE))
+	if (EnableSelfShadows && State == D3DRS_STENCILPASS && Value == D3DSTENCILOP_REPLACE)
 	{
-		if (SelfShadowTweaks &&	SH2_SpecializedLight && *SH2_SpecializedLight == 0x01)
-		{
-			Value = D3DSTENCILOP_REPLACE;
-		}
-		else
+		if (/*SelfShadowTweaks &&*/ !SH2_SpecializedLight || *SH2_SpecializedLight != 0x01)
```

Comment:
This line needs to be changed.  I it should look like the following.  We need to check if the pointer is good before using it.

```c++
if (SH2_SpecializedLight && *SH2_SpecializedLight != 0x01)
```

Also, you are removing the `SelfShadowTweaks` option but not adding it back anywhere.  Maybe you should remove this option from the settings files and replace it with a new temp option to allow enabling and disabling self shadows in specialized light.

Reply:
If `SH2_SpecializedLight` is null the if statement will short circuit and it won't dereference the pointer.

With my code if `GetSpecializedLightPointer()` fails and returns `nullptr` James will always self-shadow, with your code if it fails James will never self-shadow. These are edge cases though and yours is clearer so I'm happy to switch.

>Also, you are removing the SelfShadowTweaks option but not adding it back anywhere.

Good call, I'll add an option in my next commit.